rocksdb_admin/admin_handler.cpp

```diff
@@ -389,6 +399,50 @@ CPUThreadPoolExecutor* S3UploadAndDownloadExecutor() {
   return &executor;
 }
 
+// The dbs moved to db_tmp/ shouldnt be re-used or re-opened, so we can
+// delete them via boost filesystem operations rather than rocksdb::DestroyDB()
+void deleteDBs() {
+  static const std::string db_tmp_path = FLAGS_rocksdb_dir + "db_tmp/";
+  boost::system::error_code dir_itr_err;
+  boost::filesystem::directory_iterator itr(db_tmp_path, dir_itr_err);
+  if (dir_itr_err) {
+    LOG(ERROR) << "Can't list files in db_tmp/:" << dir_itr_err.message();
+    return;
+  }
+  for (; itr != boost::filesystem::directory_iterator(); ++itr) {
+    if (itr->path().filename() == "." || itr->path().filename() == "..") {
+      continue;
+    }
+    boost::system::error_code remove_err;
+    boost::filesystem::remove_all(itr->path(), remove_err);
+    if (remove_err) {
+      common::Stats::get()->Incr(kDeleteDBFailure);
+      LOG(ERROR) << "Cant delete db: " << itr->path() << remove_err.message();
+    } else {
+      LOG(INFO) << "Done deleting db: " << itr->path();
+    }
+    std::this_thread::sleep_for(std::chrono::seconds(kDeleteDBWaitSec));
+  }
+}
+
+// schedule the periodically health check
+void scheduleDeleteDBs() {
```

Comment:
This seems to be a little over complicated to me.
Can we just add a unique_ptr<thread> to AdminHandler and initialize it in AdminHandler:: AdminHandler if the gflag is set?

Using timer thread (delayed future) to delete data and sleep could cause trouble.